src/serializer/serializer.js

```diff
@@ -1305,12 +1329,15 @@ export class Serializer {
         }
       };
 
-      if (shouldInline || instances.length === 1 || usesArguments || anySerializedBindingModified) {
+      if (shouldInline || instances.length === 1 || usesArguments) {
         this.statistics.functionClones += instances.length - 1;
         for (let instance of instances) {
+          allocateModifiedBindingVariables.call(this, instance, names, false);
```

Comment:
Yeah, I forgot to remove this from an earlier iteration i was playing around with